subcommands/waves/init.go

```diff
@@ -126,6 +132,32 @@ func doInitWave(cmd *cobra.Command, args []string) {
 	}
 }
 
+func pruneTargets(currentTargets *client.AtsTargetsMeta, versions []string) client.AtsTargetsMeta {
+	found := []bool{}
+	keys := []string{}
+	for i, version := range versions {
+		found = append(found, false)
+		for name, file := range currentTargets.Targets {
+			custom, err := api.TargetCustom(file)
+			subcommands.DieNotNil(err)
+			if custom.Version == version {
+				keys = append(keys, name)
+				found[i] = true
+			}
+		}
+	}
+	for i, v := range found {
+		if !v {
+			subcommands.DieNotNil(fmt.Errorf("Could not find targets to prune with version = %s", versions[i]))
+		}
+	}
+	for _, name := range keys {
+		delete(currentTargets.Targets, name)
+	}
```

Comment:
It looks like you are going to the trouble of denoting all missing versions in a data structure, but then you just error out with the first missing one. I think you could simplify this logic quite a bit with something more like:
```
var missing []string
for i, version := range versions {
....
    found := false
    for name, file := range currentTargets.Targets {
        if custom.Version == version {
             delete(targets.Targets, name)
             found = true
        } 
    }
    if !found {
        missing = append(missing, version)
    } 
}
if missing != nil && len(missing) > 0 {
    die with strings.Join(missing)
}
```